### PR TITLE
[ART-6052] [WIP] update build_microshift

### PR DESCRIPTION
**Pull request by Ximinhan**
build microshift now become a regular requirement since 4.12+, and should update release.yml and attach builds to advisory automatically.

### CODE REVIEW

**Review by vfreex — COMMENTED** [review]
/hold
Needs a closer review of this process before merge.

File: pyartcd/pyartcd/pipelines/build_microshift.py
Comment (thegreyd) [inline pyartcd/pyartcd/pipelines/build_microshift.py:300]: What's the rationale to create pr and then merge immediately? Might as well do a direct commit like we do when creating advisories?

Comment (thegreyd) [reply pyartcd/pyartcd/pipelines/build_microshift.py:300]: I don't think we want `build-microshift` to be the one merging/committing, since we
could be running it manually - it would be ideal for promote job to do that 
- merge PR
- attach build
- attach bugs
- move advisory to qe

Comment (Ximinhan) [reply pyartcd/pyartcd/pipelines/build_microshift.py:300]: AFAIK direct commit is not permitted in github repo's setting ? what we do in gen-assembly is also through pr like https://github.com/openshift/ocp-build-data/pull/2613

File: pyartcd/pyartcd/pipelines/build_microshift.py
Comment (vfreex) [inline pyartcd/pyartcd/pipelines/build_microshift.py:86]: We still need aarch64 pullspec to run the rebase script.

File: pyartcd/pyartcd/pipelines/build_microshift.py
Comment (vfreex) [inline pyartcd/pyartcd/pipelines/build_microshift.py:131]: This job could be run to build microshift nightlies that are never intended to release.

File: pyartcd/pyartcd/pipelines/build_microshift.py
Comment (vfreex) [reply pyartcd/pyartcd/pipelines/build_microshift.py:300]: Good point. Merging PRs or attaching builds doesn't seem to be the responsibility of build-microshift job.
Merging PRs could be done by promote-assembly job.
Also Justin proposed another option to include a Brew event in the microshift NVR and adjust Doozer to implicitly include that build without a pin but it is up to you to think about how to implement.

**Review by vfreex — CHANGES_REQUESTED** [review]

File: pyartcd/pyartcd/pipelines/build_microshift.py
Comment (Ximinhan) [reply pyartcd/pyartcd/pipelines/build_microshift.py:300]: if we don't need to pin and just for the release process we can get the build nvr which this job triggered and directly attach to the advisory
or I saw the latest [microshift-4.12.6-202303012057.p0.g50997a2.assembly.4.12.6.el8](https://brewweb.engineering.redhat.com/brew/buildinfo?buildID=2398501) have assembly value in version

Comment (vfreex) [reply pyartcd/pyartcd/pipelines/build_microshift.py:300]: >if we don't need to pin and just for the release process we can get the build nvr which this job triggered and directly attach to the advisory
If we do like this, the builds will not be part of the assembly. That will violate the design. We need to have all builds managed by assemblies so they can be tracked and rebuilt by our toolings.

**Pull request closed without merging**
